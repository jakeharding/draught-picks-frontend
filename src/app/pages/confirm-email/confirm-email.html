<!--
  Generated template for the ConfirmEmailPage page.

  See http://ionicframework.com/docs/components/#navigation for more info on
  Ionic pages and navigation.
-->
<ion-header>
  <draught-header></draught-header>
</ion-header>


<ion-content class="ion-padding">
  <ion-row>
    <ion-col
        class="below-header ion-text-justify"
        size="12" sizeSm="10" offsetSm="1" sizeMd="8" offsetMd="2" sizeLg="6" offsetLg="3" sizeXl="4" offsetXl="4"
    >
      <div class="ion-text-center">
        <p>{{ thankYouMessage }}</p>
        <p>{{ clickMessage }}</p>
        <a class="pointer ion-text-center" (click)="goToLink()">{{ linkText }}</a>
      </div>
    </ion-col>
  </ion-row>

</ion-content>
