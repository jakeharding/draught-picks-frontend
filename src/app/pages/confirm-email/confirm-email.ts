import { Component } from '@angular/core';
import {  NavController } from '@ionic/angular';
import { BasePage } from '../BasePage';
import { UserProvider } from '../../services/user/user';
import { ToastProvider } from '../../services/toast/toast';
import { ActivatedRoute } from '@angular/router';

/**
 * Generated class for the ConfirmEmailPage page.
 *
 * See https://ionicframework.com/docs/components/#navigation for more info on
 * Ionic pages and navigation.
 */

@Component({
  selector: 'page-confirm-email',
  templateUrl: 'confirm-email.html',
  styleUrls: ['confirm-email.scss']
})
export class ConfirmEmailPage extends BasePage {
  showResendLink = false;
  linkText = 'Sign in.';
  thankYouMessage = 'Thank you for confirming your email!';
  clickMessage = 'Please click the link below to sign in.';
  confirmKey: string;
  constructor(public navCtrl: NavController,
              private route: ActivatedRoute,
              private userProvider: UserProvider,
              private toastProvider: ToastProvider) {
    super(`confirm-email/${route.snapshot.paramMap.get('key')}`);
    this.confirmKey = route.snapshot.paramMap.get('key');
  }

  ionViewDidEnter() {
      this.userProvider.confirmEmail({key: this.confirmKey}).catch(err => {
        this.showResendLink = this.isClientError(err.status);
        if (this.showResendLink) {
          this.thankYouMessage = err.error.confirm_key;
          this.clickMessage = 'Click the link below to resend the confirmation email.';
          this.linkText = 'Resend confirmation email';
        } else {
          this.toastProvider.errorToast();
          this.thankYouMessage = ToastProvider.defaultErrorMsg;
        }
        return err;
      });
  }

  goToLink() {
    let link = 'sign-in';
    if (this.showResendLink) {
      link = 'resend-email';
    }
    this.navCtrl.navigateRoot(link);
  }
}
