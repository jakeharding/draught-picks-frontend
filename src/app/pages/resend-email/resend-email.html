<!--
  Generated template for the ResendEmailPage page.

  See http://ionicframework.com/docs/components/#navigation for more info on
  Ionic pages and navigation.
-->
<ion-header>
  <draught-header></draught-header>
</ion-header>

<ion-content>
  <ion-row>
    <ion-col class="ion-text-center"
             size="12" sizeSm="10" offsetSm="1" sizeMd="6" offsetMd="3" sizeLg="6" offsetLg="3" sizeXl="4" offsetXl="4">
      <form [formGroup]="emailForm">
        <h1>Resend Email</h1>
        <br>
        <p>Please enter the email address below:</p>
        <ion-item>
          <ion-label class="ion-text-left" position="floating">Email</ion-label>
          <ion-input type="text" formControlName="email"></ion-input>
        </ion-item>
      </form>
      <ion-button
          (click)="sendConfirmEmail()"
          margin-top="20px"
          [disabled]="emailForm.invalid">
        Submit
      </ion-button>
    </ion-col>
  </ion-row>

</ion-content>
