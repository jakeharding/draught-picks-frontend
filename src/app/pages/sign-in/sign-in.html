<!--
  Generated template for the SignInPage page.

  See http://ionicframework.com/docs/components/#navigation for more info on
  Ionic pages and navigation.
-->
<ion-header>
  <draught-header></draught-header>
</ion-header>

<ion-content class="ion-padding ion-text-center">
  <ion-row>
    <ion-col size="12" sizeSm="10" offsetSm="1" sizeMd="8" offsetMd="2" sizeLg="6" offsetLg="3" sizeXl="4" offsetXl="4">
      <h1>Please Sign In</h1>
      <form [formGroup]="signInForm" (submit)="signIn()">
        <ion-item>
          <ion-label position="floating">Username or Email</ion-label>
          <ion-input tid="usernameField" formControlName="username" type="text"></ion-input>
        </ion-item>
        <br>
        <ion-item>
          <ion-label position="floating">Password</ion-label>
          <ion-input tid="passwordField" formControlName="password" type="password"></ion-input>
        </ion-item>
<!--        Work around for ion-button not submitting form when enter key is pressed-->
        <button type="submit" class="ion-page-hidden"></button>
        <ion-button
            tid="signInButton"
            (click)="signIn();"
            class="ion-margin-top"
            [disabled]="signInForm.invalid"
        >Sign In</ion-button>
        <div class="ion-text-center">
          <p>Do you not have an account?</p>
          <a routerLink="/register" routerDirection="forward">Click here to register</a>
        </div>
      </form>
    </ion-col>
  </ion-row>
</ion-content>
