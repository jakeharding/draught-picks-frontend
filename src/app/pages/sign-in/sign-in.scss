:host {
  ion-title{
    position: absolute;
    top: 0;
    left: 0;
    padding: 0 90px 1px;
    width: 100%;
    height: 100%;
    text-align: center;
    font-weight: bold;
  }
  ion-content{
    text-align: center;
  }

  a{
    cursor: pointer;
    text-decoration: none;
  }
  form{
    text-align: center;
  }
}
