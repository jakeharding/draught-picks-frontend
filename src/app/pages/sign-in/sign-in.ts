import { NavController } from '@ionic/angular';
import { Component } from '@angular/core';
import { FormBuilder, FormGroup, Validators } from '@angular/forms';
import { RegistrationPage } from '../registration/registration';
import { AuthProvider } from '../../services/auth/auth';
import { BasePage } from '../BasePage';
import { ToastProvider } from '../../services/toast/toast';

/**
 * Generated class for the SignInPage page.
 *
 * See https://ionicframework.com/docs/components/#navigation for more info on
 * Ionic pages and navigation.
 */

@Component({
  selector: 'page-sign-in',
  templateUrl: 'sign-in.html',
  styleUrls: ['./sign-in.scss']
})
export class SignInPage extends BasePage {
  signInForm: FormGroup;
  goToRegistration: any;
  goToResetPassword;
  private username: string;
  private password: string;

  constructor(public navCtrl: NavController, public formBuilder: FormBuilder, public toastProvider: ToastProvider,
              public authProvider: AuthProvider) {
    super('sign-in');
    if (this.authProvider.isLoggedIn()) {
      this.navCtrl.navigateRoot('tabs');
    }
    // this.goToRegistration = RegistrationPage;
    this.signInForm = formBuilder.group({
      username: [this.username, Validators.compose([Validators.required])],
      password: [this.password, Validators.compose([Validators.required, Validators.minLength(4)])]
    });
  }

  /**
   * signIn function
   * No Parameters
   * Gets the username and password values and validates the information then sends
   * user to the home page
   */
  public signIn() {
    this.authProvider.signIn(this.signInForm.value).then(
      (response) => {
        this.authProvider.setToken(response.token);
        this.navCtrl.navigateRoot('/tabs/home');
      },
      () => {
        this.toastProvider.errorToast('Unable to sign you in. Have you verified your email address?');
      }
    );

  }
}

