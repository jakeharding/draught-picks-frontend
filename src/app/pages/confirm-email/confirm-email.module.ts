import { NgModule } from '@angular/core';
import { IonicModule } from '@ionic/angular';
import { ConfirmEmailPage } from './confirm-email';
import { RouterModule } from '@angular/router';
import { ComponentsModule } from '../../components/components.module';

@NgModule({
  declarations: [
    ConfirmEmailPage,
  ],
  imports: [
    IonicModule,
    RouterModule.forChild([{path: '', component: ConfirmEmailPage}]),
    ComponentsModule,
  ],
})
export class ConfirmEmailPageModule {}
