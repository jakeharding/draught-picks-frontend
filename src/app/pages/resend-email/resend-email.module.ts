import { NgModule } from '@angular/core';
import { IonicModule } from '@ionic/angular';
import { ResendEmailPage } from './resend-email';
// import { ComponentsModule } from '../../components/components.module';
import { RouterModule } from '@angular/router';

@NgModule({
  declarations: [
    ResendEmailPage,
  ],
  imports: [
    IonicModule,
    RouterModule.forChild([{path: 'resent-email', component: ResendEmailPage}]),
    // ComponentsModule,
  ],
})
export class ResendEmailPageModule {}
