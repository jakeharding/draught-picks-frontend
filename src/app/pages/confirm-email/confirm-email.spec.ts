/**
 * confirm-email.spec.ts
 *
 * Created by jake
 * Created on 10/21/18
 */

import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { ConfirmEmailPage } from './confirm-email';
import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { NavController, NavParams } from '@ionic/angular';
import { UserProvider } from '../../services/user/user';
import { ToastProvider } from '../../services/toast/toast';
import { 'sign-in' } from '../sign-in/sign-in';
import { ResendEmailPage } from '../resend-email/resend-email';
import { mockNavController } from '../../../../setup-jest';

describe('ConfirmEmailPage', () => {

  let fixture: ComponentFixture<ConfirmEmailPage>;
  let component: ConfirmEmailPage;
  const navParams = new NavParams({ key: 'key'});
  const mockUserProvider = {
    confirmEmail: jest.fn(() => Promise.resolve()),
  };

  const mockToastProvider = {
    errorToast: jest.fn()
  };

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      schemas: [CUSTOM_ELEMENTS_SCHEMA],
      declarations: [ConfirmEmailPage],
      providers: [
        { provide: NavController, useValue: mockNavController},
        { provide: UserProvider, useValue: mockUserProvider },
        { provide: NavParams, useValue: navParams },
        { provide: ToastProvider, useValue: mockToastProvider}
      ]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ConfirmEmailPage);
    component = fixture.debugElement.componentInstance;
  });

  it('should initialize the ConfirmEmailPage', () => {
    expect(component).toBeTruthy();
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  test('ionViewWillEnter on failed promise with non client error', () => {
    mockUserProvider.confirmEmail.mockReturnValue(Promise.reject('error'));
    component.ionViewDidEnter();
    expect(mockUserProvider.confirmEmail).toHaveBeenCalledTimes(1);
    expect(mockUserProvider.confirmEmail).toHaveBeenCalledWith({key: 'key'});
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  test('ionViewWillEnter on failed promise with client error', () => {
    const rejectedPromise = Promise.reject({
      status: 400,
      error: {
        confirm_key: 'BOOM!'
      }
    });
    mockUserProvider.confirmEmail.mockReturnValue(rejectedPromise);
    component.ionViewDidEnter();
    expect(mockUserProvider.confirmEmail).toHaveBeenCalledTimes(1);
    expect(mockUserProvider.confirmEmail).toHaveBeenCalledWith({key: 'key'});
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  test('goToLink will go to sign-in when showResendLink is false', () => {
    component.goToLink();
    expect(mockNavController.navigateRoot).toHaveBeenCalledTimes(1);
    expect(mockNavController.navigateRoot).toHaveBeenCalledWith('sign-in');
  });

  test('goToLink will go to ResendEmailPage when showResendLink is true', () => {
    component.showResendLink = true;
    component.goToLink();
    expect(mockNavController.navigateRoot).toHaveBeenCalledTimes(1);
    expect(mockNavController.navigateRoot).toHaveBeenCalledWith('resend-email');
  });
});
