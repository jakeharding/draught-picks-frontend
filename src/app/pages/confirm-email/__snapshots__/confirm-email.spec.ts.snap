// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmEmailPage ionViewWillEnter on failed promise with client error 1`] = `
<page-confirm-email
  clickMessage={[Function String]}
  confirmKey={[Function String]}
  linkText={[Function String]}
  navCtrl={[Function Object]}
  route={[Function Object]}
  showResendLink="false"
  thankYouMessage={[Function String]}
  toastProvider={[Function Object]}
  userProvider={[Function Object]}
>
  <ion-header>
    <draught-header />
  </ion-header><ion-content
    class="ion-padding"
  >
    <ion-row>
      <ion-col
        class="below-header ion-text-justify"
        offsetlg="3"
        offsetmd="2"
        offsetsm="1"
        offsetxl="4"
        size="12"
        sizelg="6"
        sizemd="8"
        sizesm="10"
        sizexl="4"
      >
        <div
          class="ion-text-center"
        >
          <p>
            Thank you for confirming your email!
          </p>
          <p>
            Please click the link below to sign in.
          </p>
          <a
            class="pointer underline ion-text-center"
          >
            Sign in.
          </a>
        </div>
      </ion-col>
    </ion-row>
  </ion-content>
</page-confirm-email>
`;

exports[`ConfirmEmailPage ionViewWillEnter on failed promise with non client error 1`] = `
<page-confirm-email
  clickMessage={[Function String]}
  confirmKey={[Function String]}
  linkText={[Function String]}
  navCtrl={[Function Object]}
  route={[Function Object]}
  showResendLink="false"
  thankYouMessage={[Function String]}
  toastProvider={[Function Object]}
  userProvider={[Function Object]}
>
  <ion-header>
    <draught-header />
  </ion-header><ion-content
    class="ion-padding"
  >
    <ion-row>
      <ion-col
        class="below-header ion-text-justify"
        offsetlg="3"
        offsetmd="2"
        offsetsm="1"
        offsetxl="4"
        size="12"
        sizelg="6"
        sizemd="8"
        sizesm="10"
        sizexl="4"
      >
        <div
          class="ion-text-center"
        >
          <p>
            Thank you for confirming your email!
          </p>
          <p>
            Please click the link below to sign in.
          </p>
          <a
            class="pointer underline ion-text-center"
          >
            Sign in.
          </a>
        </div>
      </ion-col>
    </ion-row>
  </ion-content>
</page-confirm-email>
`;

exports[`ConfirmEmailPage should initialize the ConfirmEmailPage 1`] = `
<page-confirm-email
  clickMessage={[Function String]}
  confirmKey="undefined"
  linkText={[Function String]}
  navCtrl={[Function Object]}
  route={[Function Object]}
  showResendLink="false"
  thankYouMessage={[Function String]}
  toastProvider={[Function Object]}
  userProvider={[Function Object]}
>
  <ion-header>
    <draught-header />
  </ion-header><ion-content
    class="ion-padding"
  >
    <ion-row>
      <ion-col
        class="below-header ion-text-justify"
        offsetlg="3"
        offsetmd="2"
        offsetsm="1"
        offsetxl="4"
        size="12"
        sizelg="6"
        sizemd="8"
        sizesm="10"
        sizexl="4"
      >
        <div
          class="ion-text-center"
        >
          <p>
            Thank you for confirming your email!
          </p>
          <p>
            Please click the link below to sign in.
          </p>
          <a
            class="pointer underline ion-text-center"
          >
            Sign in.
          </a>
        </div>
      </ion-col>
    </ion-row>
  </ion-content>
</page-confirm-email>
`;
